esp/templates/program/modules/teachercheckinmodule/missingteachers.html

```diff
@@ -158,17 +158,22 @@
         <td class="checkin-column">
             {% if teacher.id not in arrived %}
             <span class="message"></span>
-            <input id="checkin_{{ teacher.username }}" name="{{ teacher.last_name }}"
+            <input id="checkin_{{ teacher.username }}" name="{{ section.parent_class.emailcode }}: {{ section.name }}"
                    class="button checkin" type="button" value="Check In" {% if when %}disabled="disabled"{% endif %} /><span />
             {% else %}
             Already checked-in
             {% endif %}
         </td>
         <td class="phone"> {{ teacher.phone }} </td>
+        <td class="text-teacher">
+            <span class="message"></span>
+            <input id="text_{{ teacher.username }}" name="{{ section.id }}"
```

Comment:
instead of using the `id` to store the username here, you can just do `data-username="{{teacher.username}}"`, and not have to do any string manipulation.  Similarly for section.  In general any `data-*` property is usable to store whatever you want, and not worry about the browser trying to assign it some other meaning -- so e.g. you could still use the `id` for CSS if desired.